### PR TITLE
return early after first path separator in splitext

### LINKED ISSUE #4944: String.splitext splits path components with dot characters
## Summary

When fed a path, the splitext method will interpret dot characters in a directory as the beginning of a file extension. If there is no extension at the end of the path, the components will be split incorrectly. This is problematic for users whose home directory structure follows the convention below:

`/Users/firstname.lastname/`

## Environment

* SuperCollider version: 3.11.0
* Operating system: macOS Catalina
* Other details (Qt version, audio driver, etc.): N/A

## Steps to reproduce

```supercollider
"foo.bar/baz".splitext
--> [ foo, bar/baz ]
```
## Expected vs. actual behavior
My expectation would be that the result of the above method call should be `[foo.bar/baz, nil]`. I originally encountered this when playing a large anonymous SynthDef, which failed with an error message about not being able to write a txarcmeta file, and realized it was trying to write it to `/Users/myFirstName`, instead of `/Users/myFirstName.LastName/Library/Application Support/Supercollider/synthdefs/`. I expect it's a potential issue elsewhere as well.

I wasn't sure whether this is a known issue or expected behavior for some reason I'm not aware of. I patched it on my machine by adding an early return in splitext, e.g.

```
splitext {
	var sep = thisProcess.platform.pathSeparator;
	block{|break|
		this.reverseDo({ arg char, i;
			// Break after the first path separator
			if ((char == sep), {break.value;});
			if (char == $\., {
				^[this.copyFromStart(this.size - 2 - i), this.copyToEnd(this.size - i)]
			});
		});
	}
	^[this, nil]
}
```

This is my first time contributing, so let me know if this is off base. If it's legit, I can open a PR.

### PR DESCRIPTION
**Pull request by frozenbears**
## Purpose and Motivation

Fixes #4944 

## Types of changes

- Bug fix

## To-do list
 Note: I verified the fix manually. I think it warrants some unit tests, especially because the method in question doesn't currently have any. But I'm new to hacking on this project so it will take a bit to get my dev environment set up.

Update: Added tests and everything is passing locally now.

- [x] Code is tested
- [x] All tests are passing
- [ ] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: SCClassLibrary/Common/Collections/String.sc [hunk 1]
```diff
@@ -429,11 +429,16 @@ String[char] : RawArray {
 		^this.primitiveFailed
 	}
 	splitext {
+		var sep = thisProcess.platform.pathSeparator;
```
Comment (brianlheim) [inline SCClassLibrary/Common/Collections/String.sc:432]: please use `Platform:isPathSeparator` instead. we support using either `/` or `\` as a directory separator on windows, but `Platform:pathSeparator` is only `\` on Windows, so this code would still consider `foo.bar/baz` to have the extension `bar/baz` there. sorry, neither of these things is documented with that information.

Comment (marc-scig) [reply SCClassLibrary/Common/Collections/String.sc:432]: cool, I'll do that

**Comment by brianlheim** [comment]
hi @frozenbears, are you still interested in this PR? it would be a huge help if you could fix it up!

**Comment by frozenbears** [comment]
Hi @brianlheim, sorry to leave you hanging, things got a little crazy around here recently. I'm still interested, and should have an update soon.

**Comment by frozenbears** [comment]
Oh, you're right. One sec.

**Pull request merged**